### PR TITLE
Add the "patch-steps" proposal

### PR DESCRIPTION
**Pull request by 20kdc**
A proposal for a new JSON patch format that fits the style of other game formats.
For further details, please see the file itself.

### CODE REVIEW

File: proposals/1/patch-steps.md
Comment (2767mr) [inline proposals/1/patch-steps.md:19]: Less likely to break as compared to what method?

File: proposals/1/patch-steps.md
Comment (2767mr) [inline proposals/1/patch-steps.md]: Maybe specify that src is CC root (or assets root?)

**Review by 2767mr — CHANGES_REQUESTED** [review]
Only 2 small things.
It would be great if TS declarations of the steps were added, although not necessary for the proposal

File: proposals/1/patch-steps.md
Comment (20kdc) [reply proposals/1/patch-steps.md:19]: As compared to the object-based method. As noted previously it works for most stuff, but specifically not arrays.

File: proposals/1/patch-steps.md
Comment (20kdc) [reply proposals/1/patch-steps.md]: Kind of a complex question by itself, since the format part's theoretically supposed to be partially 'independent' to what's loading "src"... I'll work out how to clarify this bit after I've dealt with standardized-mods

Comment (20kdc) [reply proposals/1/patch-steps.md]: ...Wait, I actually did handle this in the spec - on the next line.
`JSON is loaded from the file specified with "src" (the value does *not* contain ig.root, that's prepended as usual).`
EDIT: I'll add an example to clarify

**Comment by 20kdc** [comment]
Probably best not to merge given the consideration of the more standard JSON Patch format

**Comment by 20kdc** [comment]
Nevermind? I'm not sure if this should or should not be gone with
We do need *a* solution for the issue though, ItemAPI is kinda useless without it

**Comment by 20kdc** [comment]
CCItemTest at:
https://github.com/20kdc/CLS-20kdc-Code/tree/master/mods
serves as a test mod for this

Should be implemented as written by https://github.com/CCDirectLink/CCLoader/pull/41 for CCLoader.

**Comment by 20kdc** [comment]
Given this is now an implemented cross-modloader standard, it should probably be merged?

**Review by 2767mr — DISMISSED** [review]

**Comment by 20kdc** [comment]
On hold because of the URL-style paths thing, along with some other minor convenience features

**Comment by 20kdc** [comment]
Not on hold anymore, but before merging, @2767mr, @ac2pic, there's a new version of the library linked that implements the stuff mentioned.

File: proposals/1/patch-steps.md
Comment (2767mr) [inline proposals/1/patch-steps.md]: Please add ts syntax highlighting

**Review by 2767mr — CHANGES_REQUESTED** [review]
I don't really get why the md description of the patch steps were removed but it works as it is now.

**Review by 2767mr — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**
